src/others/helpers/aid-request-helpers.ts

```diff
@@ -87,3 +87,20 @@ export const groupedByCategoriesToTableData = ({ category_id, total, aidRequests
 
 const totalCalculator = (aidRequests: AidRequest[]): number =>
   aidRequests.reduce((sum, aidRequest) => sum + aidRequest.requested_amount, 0);
+
+export type RegionData = { [id: string]: number };
+
+export const groupByRegions = (aidRequests: AidRequest[], translateLocation: (city_id: number) => Location | undefined): RegionData => {
+  const regionToCount: RegionData = {};
+  aidRequests.forEach((req) => {
+    const city = translateLocation(req.city_id);
```

Comment:
**question**

Why do you need to translate to get the `region_id`?

This implementation is still very much imperative, but let's leave it for now.

Reply:
city_id is a property on AidRequest groupedByCityId, or on any other AidRequest under .aidRequests

The translateLocation adds tranlated name and coords.

Reply:
translateLocation returns all metadata associated with a city, that is id, name, lat/lon AND region_id.

Reply:
i assume this is resolved